modules/apps/object/object-test-util/src/main/java/com/liferay/object/test/util/TreeTestUtil.java

```diff
@@ -286,7 +287,21 @@ private static void _assertTree(
 					node.getChildNodes(), unsafeFunction, String.class));
 		}
 
-		AssertUtils.assertEquals(expectedMap, actualMap);
+		Assert.assertEquals(
+			"The maps have different sizes", expectedMap.size(),
+			actualMap.size());
+
+		for (Map.Entry<String, String[]> entry : expectedMap.entrySet()) {
+			String[] expectedValues = entry.getValue();
+
+			Arrays.sort(expectedValues);
+
+			String[] actualValues = actualMap.get(entry.getKey());
+
+			Arrays.sort(actualValues);
+
+			Assert.assertArrayEquals(expectedValues, actualValues);
+		}
```

Comment:
@seltong can we simplify to:

```
for (String key : expectedMap.keySet()) {
	String[] actualValues = actualMap.get(key);
	String[] expectedValues = expectedMap.get(key);

	Assert.assertEquals(expectedValues.length, actualValues.length);
	Assert.assertTrue(ArrayUtil.containsAll(expectedValues, actualValues));
}
```
?

This way we don't have to change both arrays with `Arrays.sort`.

Reply:
The SF will fail because we cannot use the `String key : expectedMap.keySet()` in this case. (see the sf error message below)

```
Use Map.entrySet() to retrieve values, see 
https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/map_iteration_check.markdown: 
./modules/apps/object/object-test-util/src/main/java/com/liferay/object/test/util/TreeTestUtil.java 291 (Checkstyle:MapIterationCheck)
```

The SF also suggested adding the `Arrays.toString(actualValues)` to the `Assert.assertEquals()`.

So the closest solution I could come up with based on your suggestion was:

```
Assert.assertEquals(
	"The maps have different sizes", expectedMap.size(),
	actualMap.size());

for (Map.Entry<String, String[]> entry : expectedMap.entrySet()) {
	String[] actualValues = actualMap.get(entry.getKey());
	String[] expectedValues = entry.getValue();

	Assert.assertEquals(
		Arrays.toString(actualValues), expectedValues.length,
		actualValues.length);
	Assert.assertTrue(
		ArrayUtil.containsAll(expectedValues, actualValues));
}
```

However, I still worry about the order of `actualValues` and `expectedValues`. Following the order of use, `actualValues` should come first...